docs/source/existing_code.rst

```diff
@@ -0,0 +1,174 @@
+annotations requested by mypy, or by adding ``# type: ignore``
+comments to silence errors you don't want to fix now.
+
+In particular, mypy often generates errors about modules that it can't
+find or that don't have stub files:
+
+.. code-block:: text
+
+    core/config.py:7: error: Cannot find module named 'frobnicate'
+    core/model.py:9: error: Cannot find module named 'acme'
+    ...
+
+This is normal, and you can easily ignore these errors. For example,
+here we ignore an error about a third-party module ``frobnicate`` that
+doesn't have stubs using ``# type: ignore``:
+
+.. code-block:: python
+
+   import frobnicate  # type: ignore
+   ...
+   frobnicate.initialize()  # OK (but not checked)
+
+You can also use a mypy configuration file, which is convenient if
+there are a large number of errors to ignore. For example, to disable
+errors about importing ``frobnicate`` and ``acme`` everwhere in your
+codebase, use a config like this:
+
+.. code-block:: text
+
+   [mypy-frobnicate.*]
+   ignore_missing_imports = True
+
+   [mypy-acme.*]
+   ignore_missing_imports = True
+
+You can add multiple sections for different modules that should be
+ignored.
+
+If your config file is named ``mypy.ini``, this is how you run mypy:
+
+.. code-block:: text
+
+   mypy --config-file mypy.ini mycode/
+
+If you get a large number of errors, you may want to ignore all errors
+about missing imports.  This can easily cause problems later on and
+hide real errors, and it's only recommended as a last resort. We
+don't explain this option in more detail here.
+
+Mypy follows imports by default. This can result in a few files passed
+on the command line causing mypy to process a large number of imported
+files, resulting in lots of errors you don't want to deal with at the
+moment. There is a config file option to disable this behavior, but
+since this can hide errors, it's not recommended for most users.
+
+Mypy runner script
+------------------
+
+Introduce a mypy runner script that runs mypy, so that every developer
+will use mypy consistently. Here are some things you may want to do in
+the script:
+
+* Ensure that the correct version of mypy is installed.
+
+* Specify mypy config file or command-line options.
+
+* Provide set of files to type check. You may want to implement
+  inclusion and exclusion filters for full control of the file
+  list.
+
+Continuous Integration
+----------------------
+
+Once you have a clean mypy run and a runner script for a part
+of your codebase, set up your Continuous Integration (CI) system to
+run mypy to ensure that developers won't introduce bad annotations.
+A simple CI script could look something like this:
+
+.. code-block:: text
+
+    python3 -m pip install mypy==0.600  # Pinned version avoids surprises
+    scripts/mypy  # Runs with the correct options
+
+Annotate library modules
+------------------------
+
+Most projects have some commonly used library code. It's a good idea
```

Comment:
I'm a little worried that people might think "library code" == "third-party libraries" and get the impression we're asking them to go off and annotate Django or something. Not sure what the best way to clarify this is though (or if it's even worth clarifying).